components/core/src/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -532,37 +513,28 @@ namespace ffi::ir_stream {
         return IRErrorCode_Success;
     }
 
-    namespace four_byte_encoding {
-        IRErrorCode decode_preamble (IrBuffer& ir_buf, TimestampInfo& ts_info,
-                                     epoch_time_ms_t& reference_ts)
-        {
-            ir_buf.init_internal_pos();
-
-            string_view json_metadata;
-            if (auto error_code = read_json_metadata(ir_buf, json_metadata);
-                    error_code != IRErrorCode_Success)
-            {
-                return error_code;
-            }
-
-            try {
-                auto metadata_json = nlohmann::json::parse(json_metadata);
-                const auto& version = metadata_json.at(cProtocol::Metadata::VersionKey);
-                if (version != cProtocol::Metadata::VersionValue) {
-                    return IRErrorCode_Unsupported_Version;
-                }
+    IRErrorCode decode_preamble (IrBuffer& ir_buf, encoded_tag_t& metadata_type,
```

Comment:
is there a reason why the function doesn't simply return the metadata as a string by reference?